programs/validator-history/src/state.rs

```diff
@@ -148,6 +180,69 @@ impl CircBuf {
     pub fn arr_mut(&mut self) -> &mut [ValidatorHistoryEntry] {
         &mut self.arr
     }
+
+    pub fn epoch_range(
+        &self,
+        start_epoch: u16,
+        end_epoch: u16,
+    ) -> Option<Vec<&ValidatorHistoryEntry>> {
```

Comment:
another option: Vec<Option<&ValidatorHistoryEntry>>. 

what if epoch missing? for averaging, i ask for 5, get 2. i assume you use x.epoch_range(...).len() for avg?